PACKAGE = "l3cam_ros"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# Parameters
# gen.add(name, paramtype, level, description, default, min, max, edit_method)

# CHANGE_THERMAL_CAMERA_COLORMAP
change_thermal_camera_colormap_enum = gen.enum([gen.const("thermal_WHITE",      int_t, 1,   "White"),
                                                gen.const("thermal_BLACK",      int_t, 17,  "Black"),
                                                gen.const("thermal_IRON",       int_t, 20,  "Iron"),
                                                gen.const("thermal_COOL",       int_t, 2,   "Cool"),
                                                gen.const("thermal_AMBER",      int_t, 9,   "Amber"),
                                                gen.const("thermal_INDIGO",     int_t, 10,  "Indigo"),
                                                gen.const("thermal_TYRIAN",     int_t, 16,  "Tyrian"),
                                                gen.const("thermal_GLORY",      int_t, 8,   "Glory"),
                                                gen.const("thermal_ENVY",       int_t, 16,  "Envy"),
                                                gen.const("thermal_WHITE_NEW",  int_t, 100, "White new"),
                                                gen.const("thermal_BLACK_NEW",  int_t, 101, "Black new"),
                                                gen.const("thermal_SPECTRA",    int_t, 102, "Spectra"),
                                                gen.const("thermal_PRISM",      int_t, 103, "Prism"),
                                                gen.const("thermal_TYRIAN_NEW", int_t, 104, "Tyrian new"),
                                                gen.const("thermal_AMBER_NEW",  int_t, 105, "Amber new"),
                                                gen.const("thermal_IRON_NEW",   int_t, 106, "Iron new"),
                                                gen.const("thermal_HI",         int_t, 107, "Hi"),
                                                gen.const("thermal_HILO",       int_t, 108, "HiLo")],
                                               "Enum to set thermal camera colormap")
gen.add("change_thermal_camera_colormap", int_t, 0, "Change thermal camera colormap", 1, 1, 108, edit_method=change_thermal_camera_colormap_enum)
# ENABLE_THERMAL_CAMERA_TEMPERATURE_FILTER
gen.add("enable_thermal_camera_temperature_filter", bool_t, 1, "Enable thermal camera temperature filter", False)
# CHANGE_THERMAL_CAMERA_TEMPERATURE_FILTER
gen.add("change_thermal_camera_temperature_filter_min", int_t, 2, "Change thermal camera max temperature filter", 0,  -40, 200)
gen.add("change_thermal_camera_temperature_filter_max", int_t, 3, "Change thermal camera min temperature filter", 50, -40, 200)

exit(gen.generate(PACKAGE, "l3cam_ros", "ThermalCamera"))
